### ISSUE TITLE
Evaluating on the test set every training epoch

### ISSUE THREAD

**Issue by maeyen** [mingyangliu1024/ACON#5]
https://github.com/mingyangliu1024/ACON/blob/ac545487b69cf0ba5b1219ef425010898ff186c1/trainers/da_trainer.py#L184-L219  Confused about Using 'trg_test_dl'( which is used as test set  for final testing ) as the validation set to select the best model at training time
`
                    acc, f1 = self.evaluate()
                    self.logger.debug('acc {}   f1 {}'.format(acc, f1))
                    if f1>=self.best_f1:
                        self.best_f1 = f1
                        self.logger.debug('best model {}'.format(epoch))
                        algorithm.save_model(self.model_path)`

`def evaluate(self, final=False, data='t'):
        assert data in ['t', 's']
        self.algorithm.eval()
        if final == True:
            self.algorithm.load_model(self.model_path)
    
        self.trg_pred_labels = np.array([])
        self.trg_true_labels = np.array([])

        with torch.no_grad():
            if data == 't':
                dataloader = self.trg_test_dl
            elif data == 's':
                dataloader = self.src_test_dl

            for data, labels in dataloader:
                data = data.float().to(self.device)
                labels = labels.view((-1)).long().to(self.device)

                predictions = self.algorithm.predict(data)

                # compute loss
                pred = predictions.detach().argmax(dim=1)  # get the index of the max log-probability

                self.trg_pred_labels = np.append(self.trg_pred_labels, pred.cpu().numpy())
                self.trg_true_labels = np.append(self.trg_true_labels, labels.data.cpu().numpy())
        
        accuracy = accuracy_score(self.trg_true_labels, self.trg_pred_labels)
        f1 = f1_score(self.trg_pred_labels, self.trg_true_labels, pos_label=None, average="macro")
        return accuracy*100, f1
`
